### PR TITLE
Feature: Substrate 1.1.0

### PR DESCRIPTION
**Pull request by rakanalh**
### Description

Substrate [v1.1.0](https://github.com/paritytech/polkadot-sdk/releases/tag/polkadot-v1.1.0) upgrade which is required for some OpenGov migrations.

### Types of Changes

- [ ] Tech Debt (Code improvements)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Dependency upgrade (A change in substrate or any 3rd party crate version)

### Migrations and Hooks

- [x] This change requires a runtime migration.
- [ ] Modifies `on_initialize`
- [ ] Modifies `on_finalize`

### Checklist

- [x] Change has been tested locally.
- [ ] Change adds / updates tests.
- [x] Changelog doc updated.

### TODO
- [x] Update dependencies
- [x] Cleanup existing migrations
- [x] Include new migrations.

### CODE REVIEW

File: node/service/src/lib.rs [hunk 1]
```diff
@@ -161,8 +162,7 @@ where
 		+ Send
 		+ Sync
 		+ 'static,
-	RuntimeApi::RuntimeApi:
-		RuntimeApiCollection<StateBackend = sc_client_api::StateBackendFor<FullBackend, Block>>,
```
Comment (yahortsaryk) [inline node/service/src/lib.rs:165]: This change comes from substrate [PR #14612](https://github.com/paritytech/substrate/pull/14612)

File: node/service/src/lib.rs [hunk 2]
```diff
@@ -53,8 +56,7 @@ where
...
```
Comment (yahortsaryk) [inline node/service/src/lib.rs:57]: This change comes from substrate [PR #14612](https://github.com/paritytech/substrate/pull/14612)

File: node/service/src/lib.rs [hunk 3]
```diff
@@ -130,7 +131,7 @@ fn new_partial<RuntimeApi, ExecutorDispatch>(
 		FullClient<RuntimeApi, ExecutorDispatch>,
 		FullBackend,
 		FullSelectChain,
-		sc_consensus::DefaultImportQueue<Block, FullClient<RuntimeApi, ExecutorDispatch>>,
```
Comment (yahortsaryk) [inline node/service/src/lib.rs:133]: This change comes from substrate [PR #14612](https://github.com/paritytech/substrate/pull/14612)

File: node/service/src/lib.rs [hunk 4]
```diff
@@ -72,8 +74,7 @@ where
 		+ Send
 		+ Sync
 		+ 'static,
-	RuntimeApi::RuntimeApi:
-		RuntimeApiCollection<StateBackend = sc_client_api::StateBackendFor<FullBackend, Block>>,
```
Comment (yahortsaryk) [inline node/service/src/lib.rs:76]: This change comes from substrate [PR #14612](https://github.com/paritytech/substrate/pull/14612)

File: node/service/src/lib.rs (hunk 1 above)
Comment (yahortsaryk) [inline node/service/src/lib.rs:165]: This change comes from substrate [PR #14612](https://github.com/paritytech/substrate/pull/14612)

File: node/service/src/lib.rs [hunk 5]
```diff
@@ -340,11 +342,10 @@ where
...
```
Comment (yahortsaryk) [inline node/service/src/lib.rs:344]: This change comes from substrate [PR #14612](https://github.com/paritytech/substrate/pull/14612)

File: node/service/src/lib.rs [hunk 6]
```diff
@@ -583,10 +584,9 @@ impl ExecuteWithClient for RevertConsensus {
 
 	fn execute_with_client<Client, Api, Backend>(self, client: Arc<Client>) -> Self::Output
 	where
-		<Api as sp_api::ApiExt<Block>>::StateBackend: sp_api::StateBackend<BlakeTwo256>,
-		Backend: sc_client_api::Backend<Block> + 'static,
+		Backend: sc_client_api::Backend<Block>,
 		Backend::State: sp_api::StateBackend<BlakeTwo256>,
-		Api: RuntimeApiCollection<StateBackend = Backend::State>,
```
Comment (yahortsaryk) [inline node/service/src/lib.rs:589]: This change comes from substrate [PR #14612](https://github.com/paritytech/substrate/pull/14612)

File: node/service/src/lib.rs [hunk 7]
```diff
@@ -583,10 +584,9 @@ impl ExecuteWithClient for RevertConsensus {
 
 	fn execute_with_client<Client, Api, Backend>(self, client: Arc<Client>) -> Self::Output
 	where
-		<Api as sp_api::ApiExt<Block>>::StateBackend: sp_api::StateBackend<BlakeTwo256>,
```
Comment (yahortsaryk) [inline node/service/src/lib.rs:586]: This change comes from substrate [PR #14612](https://github.com/paritytech/substrate/pull/14612)

File: node/service/src/lib.rs [hunk 8]
```diff
@@ -610,12 +610,7 @@ macro_rules! chain_ops {
 pub fn new_chain_ops(
 	config: &Configuration,
 ) -> Result<
-	(
-		Arc<Client>,
-		Arc<FullBackend>,
-		sc_consensus::BasicQueue<Block, PrefixedMemoryDB<BlakeTwo256>>,
```
Comment (yahortsaryk) [inline node/service/src/lib.rs:616]: This change comes from substrate [PR #14612](https://github.com/paritytech/substrate/pull/14612)

File: node/service/src/lib.rs [hunk 9]
```diff
@@ -38,6 +37,10 @@ pub use sc_service::ChainSpec;
 use sc_transaction_pool_api::OffchainTransactionPoolFactory;
 pub use sp_api::ConstructRuntimeApi;
 
+/// The minimum period of blocks on which justifications will be
+/// imported and generated.
+const GRANDPA_JUSTIFICATION_PERIOD: u32 = 512;
```
Comment (yahortsaryk) [inline node/service/src/lib.rs:42]: This change comes from substrate [PR #14423](https://github.com/paritytech/substrate/pull/14423)

File: node/service/src/lib.rs [hunk 10]
```diff
@@ -532,7 +532,7 @@ where
 	let config = sc_consensus_grandpa::Config {
 		// FIXME #1578 make this available through chainspec
 		gossip_duration: std::time::Duration::from_millis(333),
-		justification_period: 512,
+		justification_generation_period: GRANDPA_JUSTIFICATION_PERIOD,
```
Comment (yahortsaryk) [inline node/service/src/lib.rs:535]: This change is introduced by substrate [PR #14423](https://github.com/paritytech/substrate/pull/14423)

File: node/service/src/lib.rs [hunk 11]
```diff
@@ -174,13 +174,14 @@ where
 		task_manager.spawn_essential_handle(),
 		client.clone(),
 	);
-
 	let (grandpa_block_import, grandpa_link) = sc_consensus_grandpa::block_import(
 		client.clone(),
+		GRANDPA_JUSTIFICATION_PERIOD,
```
Comment (yahortsaryk) [inline node/service/src/lib.rs:179]: This change is introduced by substrate [PR #14423](https://github.com/paritytech/substrate/pull/14423)

File: node/client/src/lib.rs [hunk 12]
```diff
@@ -128,10 +128,9 @@ pub trait ExecuteWithClient {
 	/// Execute whatever should be executed with the given client instance.
 	fn execute_with_client<Client, Api, Backend>(self, client: Arc<Client>) -> Self::Output
 	where
-		<Api as sp_api::ApiExt<Block>>::StateBackend: sp_api::StateBackend<BlakeTwo256>,
```
Comment (yahortsaryk) [inline node/client/src/lib.rs:131]: This change is introduced by substrate [PR #14612](https://github.com/paritytech/substrate/pull/14612)

File: node/client/src/lib.rs [hunk 13]
```diff
@@ -483,13 +482,10 @@ pub trait RuntimeApiCollection:
 	+ sp_offchain::OffchainWorkerApi<Block>
 	+ sp_session::SessionKeys<Block>
 	+ sp_authority_discovery::AuthorityDiscoveryApi<Block>
-where
-	<Self as sp_api::ApiExt<Block>>::StateBackend: sp_api::StateBackend<BlakeTwo256>,
```
Comment (yahortsaryk) [inline node/client/src/lib.rs:487]: This change is introduced by substrate [PR #14612](https://github.com/paritytech/substrate/pull/14612)

File: node/client/src/lib.rs [hunk 14]
```diff
@@ -500,8 +496,7 @@ where
 		+ sp_api::Metadata<Block>
 		+ sp_offchain::OffchainWorkerApi<Block>
 		+ sp_session::SessionKeys<Block>
-		+ sp_authority_discovery::AuthorityDiscoveryApi<Block>,
-	<Self as sp_api::ApiExt<Block>>::StateBackend: sp_api::StateBackend<BlakeTwo256>,
```
Comment (yahortsaryk) [inline node/client/src/lib.rs:504]: This change is introduced by substrate [PR #14612](https://github.com/paritytech/substrate/pull/14612)

File: node/client/src/lib.rs [hunk 15]
```diff
@@ -72,7 +72,7 @@ where
 	Block: BlockT,
 	Backend: BackendT<Block>,
 	Backend::State: sp_api::StateBackend<BlakeTwo256>,
-	Self::Api: RuntimeApiCollection<StateBackend = Backend::State>,
```
Comment (yahortsaryk) [inline node/client/src/lib.rs:75]: This change is introduced by substrate [PR #14612](https://github.com/paritytech/substrate/pull/14612)

File: runtime/cere-dev/src/lib.rs [hunk 16]
```diff
@@ -1033,11 +1042,15 @@ impl pallet_contracts::Config for Runtime {
...
+	type Debug = ();
+	type Environment = ();
 	type Migrations = (
-		cere_runtime_common::migrations::ContractsMigration<Runtime>,
-		// pallet_contracts::migration::v10::Migration<Runtime>,
-		pallet_contracts::migration::v11::Migration<Runtime>,
-		pallet_contracts::migration::v12::Migration<Runtime>,
+		pallet_contracts::migration::v13::Migration<Runtime>,
+		pallet_contracts::migration::v14::Migration<Runtime, Balances>,
+		pallet_contracts::migration::v15::Migration<Runtime>,
```
Comment (yahortsaryk) [inline runtime/cere-dev/src/lib.rs:1053]: This change is introduced by [PR #14589](https://github.com/paritytech/substrate/pull/14589)

**Review by yahortsaryk — APPROVED** [review]

**Review by yahortsaryk — APPROVED** [review]

**Review by MRamanenkau — APPROVED** [review]

**Review by yahortsaryk — APPROVED** [review]

**Pull request merged**